moveit_core/trajectory_processing/src/cartesian_speed.cpp

```diff
@@ -0,0 +1,135 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Authors: Benjamin Scholz, Thies Oelerich */
+
+#include <moveit/trajectory_processing/cartesian_speed.h>
+
+// Name of logger
+const char* LOGGER_NAME = "trajectory_processing.cartesian_speed";
+
+namespace trajectory_processing
+{
+bool setMaxCartesianLinkSpeed(robot_trajectory::RobotTrajectory& trajectory, const double max_speed,
+                              std::string link_name)
+{
+  // In case the link name is not set, retrieve an end effector name from the
+  // joint model group specified in the robot trajectory
+  if (link_name.empty())
+  {
+    std::vector<std::string> tips;
+    trajectory.getGroup()->getEndEffectorTips(tips);
+    if (tips.empty())
+    {
+      ROS_ERROR_STREAM_NAMED(LOGGER_NAME, "No end effector defined for group attached to trajectory.");
+      return false;
+    }
+    link_name = tips[0];
+    if (tips.size() > 1)
+      ROS_INFO_STREAM_NAMED(LOGGER_NAME,
+                            "More than one end effector found, using first one which is '" << link_name << "'");
+  }
+
+  // Get link model based on given end link
+  if (!trajectory.getGroup()->hasLinkModel(link_name))
+    ROS_ERROR_STREAM_NAMED(LOGGER_NAME, "Link model was not specified in the robot trajectory");
+  const moveit::core::LinkModel* link_model = trajectory.getGroup()->getLinkModel(link_name);
+
+  // Call function for speed setting using the created link model
+  return setMaxCartesianLinkSpeed(trajectory, max_speed, link_model);
+}
+
+bool setMaxCartesianLinkSpeed(robot_trajectory::RobotTrajectory& trajectory, const double max_speed,
```

Comment:
```suggestion
bool limitMaxCartesianLinkSpeed(robot_trajectory::RobotTrajectory& trajectory, const double max_speed,
```
This has the same name as the MGI function, but it actually does something else. What about this name, so the effect is clearer?